### PR TITLE
sync player grabbing vines

### PR DESCRIPTION
**Pull request by youbitchoc**
doesn't sync the position of the vine itself, which can get noticeably desynced

also change chunkPosLeniency to a bool in opo that skips chunk sync entirely rather than relying on a threshold, as suggested by henpe

(the diff looks weird because i unindented some stuff)

### CODE REVIEW

**Comment by youbitchoc** [comment]
we should probably sync untethered climbables too (specifically those coral neuron thingies in pebbles) but that'll require a new onlineentity? so probably a separate PR

File: Extensions.cs [hunk 1]
```diff
@@ -216,5 +217,14 @@ public static Color SafeColorRange(this Color valuecolor)
...
         }
+
+        public static ILCursor GotoNextOutEnd(this ILCursor cursor, out ILLabel end, params Func<Mono.Cecil.Cil.Instruction, bool>[] predicates)
+        {
+            var ret = cursor.GotoNext(moveType: MoveType.AfterLabel, predicates);
+            var c = cursor.Clone();
+            c.Index += predicates.Length;
+            end = c.MarkLabel();
+            return ret;
+        }
```
Comment (henpemaz) [inline Extensions.cs:228]: 🤨 

I see the attempt at making a pattern out of these types of matches, but 1. it's not a lot of code; 2. they'll all have small nuances specially with how if/else blocks might be organized; and lastly 3. I wouldn't encourage matching the whole body of an if, matches are ideally as short as possible so to not be affected by other mods' edits

**Review by henpemaz — APPROVED** [review]
Haven't checked the whole diff but the stuff thats new looks alright, couple notes:

as is oc.lenientpos might be a tick late (inc. stuff sets it to true but base already ran), not sure if intented, not sure if might have any meaningful side-effects

that `oc.lenientPos = false;` might get moved around later when more things start using this system, but no worries for now

**Comment by youbitchoc** [comment]
> oc.lenientpos might be a tick late

split it out into a separate function so hopefully base.readto won't be too cluttered

**Review by henpemaz — APPROVED** [review]

**Pull request merged**